kentikapi/internal/api_payloads/saved_filter.go

```diff
@@ -79,16 +80,46 @@ func SavedFilterToPayload(sf models.SavedFilter) SavedFilterPayload {
 	}
 }
 
-type FiltersPayload struct {
+func SavedFilterToCreatePayload(sf models.SavedFilter) CreateSavedFilterRequest {
+	return CreateSavedFilterRequest{
+		savedFilterPayload: savedFilterPayload{
+			ID:                IntAsString(sf.ID),
+			CompanyID:         sf.CompanyID,
+			FilterName:        sf.FilterName,
+			FilterDescription: sf.FilterDescription,
+			FilterLevel:       sf.FilterLevel,
+			CreatedDate:       &sf.CreatedDate,
+			UpdatedDate:       &sf.UpdatedDate,
+			Filters:           FiltersToPayload(sf.Filters),
+		},
+	}
+}
+
+func SavedFilterToUpdatePayload(sf models.SavedFilter) UpdateSavedFilterRequest {
+	return UpdateSavedFilterRequest{
+		savedFilterPayload: savedFilterPayload{
+			ID:                IntAsString(sf.ID),
+			CompanyID:         sf.CompanyID,
+			FilterName:        sf.FilterName,
+			FilterDescription: sf.FilterDescription,
+			FilterLevel:       sf.FilterLevel,
+			CreatedDate:       &sf.CreatedDate,
+			UpdatedDate:       &sf.UpdatedDate,
+			Filters:           FiltersToPayload(sf.Filters),
+		},
+	}
+}
+
+type filtersPayload struct {
 	Connector    string                `json:"connector"`
 	Custom       *bool                 `json:"custom,omitempty"`
-	FilterGroups []FilterGroupsPayload `json:"filterGroups"`
+	FilterGroups []filterGroupsPayload `json:"filterGroups" response:"get,post,put"`
```

Comment:
Does response validation do anything here? See: https://github.com/kentik/community_sdk_golang/pull/16#discussion_r591681147

Reply:
I think it's unnecessary here and I will remove it